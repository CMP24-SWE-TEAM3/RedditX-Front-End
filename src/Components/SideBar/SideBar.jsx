// import Components
import React from "react";
import {OffcanvasBody} from "./SideBar.styled"
import Offcanvas from "react-bootstrap/Offcanvas";
import {AiOutlineContainer} from "react-icons/ai";
import {ButtonStyled} from "./SideBar.styled";
import DropDownItem from "../DropDownItem/DropDownItem";
import {GrFormPrevious} from "react-icons/gr";

/**
 * Component that displays a sidebar which includes a dropdown menu in it
 * @returns {React.Component}
 */
const SideBar = ({showButton, showSideBar, setShowSideBar}) => {

    const handleClose = () => setShowSideBar(false);
    const handleShow = () => {
        setShowSideBar(true);
    };

    return (
        <>
            {showButton && <ButtonStyled onClick={handleShow}>
                <div id={'side-bar-action'} className={'open-side'}>
                    <AiOutlineContainer/><GrFormPrevious/>
                </div>

            </ButtonStyled>}

            <OffcanvasBody id={'left-sidebar'} show={showSideBar} onHide={handleClose} className={'mt-5'} backdrop={false}>
                <Offcanvas.Body>
                    <Offcanvas.Header closeButton>
                        <Offcanvas.Title/>
                    </Offcanvas.Header>
                    <DropDownItem/>
                </Offcanvas.Body>
            </OffcanvasBody>
        </>
    );
}

export default SideBar;