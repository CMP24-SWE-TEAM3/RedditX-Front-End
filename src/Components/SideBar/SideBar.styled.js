import styled from "styled-components";
import Button from "react-bootstrap/Button";
import Offcanvas from "react-bootstrap/Offcanvas";

export const ButtonStyled = styled(Button)`
  @media (max-width: 1530px) {
    display: none;
  }
  width: 10px;
  background: none;
  color: ${({theme}) => theme.color.primary};
  border: none;

  :hover, :active, :focus-visible {
    background: none !important;
    color: ${({theme}) => theme.color.primary} !important;
    box-shadow: none;
  }

  .open-side {
    margin-left:110px;
  }

`
// canvas styles
export const OffcanvasBody = styled(Offcanvas)`
  .offcanvas-body {
    overflow-y: scroll;
  }
  width: 270px !important;
  position: absolute;
  background: ${({theme}) => theme.background.primary};

  .btn-close:focus {
    box-shadow: none !important;
  }

  .btn-close {
    color: ${({theme}) => theme.color.primary} !important;
  }
`