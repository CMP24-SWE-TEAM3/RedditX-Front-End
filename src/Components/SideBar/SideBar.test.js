import {shallow} from 'enzyme';

// import Components
import SideBar from "./SideBar";

describe("Side Bar Component", () => {
    it("this is a test for Side Bar", () => {
        expect(shallow(<SideBar/>)).toMatchSnapshot();

    });
});