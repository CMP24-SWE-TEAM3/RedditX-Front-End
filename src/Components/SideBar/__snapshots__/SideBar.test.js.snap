// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Bar Component this is a test for Side Bar 1`] = `ShallowWrapper {}`;
